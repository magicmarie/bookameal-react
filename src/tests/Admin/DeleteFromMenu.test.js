import React from "react";
import { shallow } from "enzyme";
import DeleteMealFromMenu from "../../components/Admin/DeleteFromMenu";

describe("DeleteMealFromMenuCard component", () => {
  let wrapper;
  beforeEach(() => {
    const ConfirmDeleteMeal=jest.fn()
    wrapper = shallow(<DeleteMealFromMenu ConfirmDeleteMeal={ConfirmDeleteMeal}/>);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should confirm delete on click", () => {
    const handleEvent = jest.fn()
    wrapper.find('button.btn-danger').simulate('click', handleEvent)
  });
});
