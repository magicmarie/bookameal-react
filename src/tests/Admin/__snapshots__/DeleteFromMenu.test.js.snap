// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMealFromMenuCard component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteMealFromMenu
    ConfirmDeleteMeal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal fade"
        id="deleteFromMenuModalundefined"
>
        <div
                className="modal-dialog modal-confirm"
        >
                <div
                        className="modal-content"
                >
                        <div
                                className="modal-header"
                        >
                                <h4
                                        className="modal-title"
                                >
                                        Are you sure?
                                </h4>
                                <Unknown />
                        </div>
                        <div
                                className="modal-body"
                        >
                                <p>
                                        Do you really want to delete 
                                        <strong />
                                         
                                        from the menu? This process cannot be undone.
                                </p>
                        </div>
                        <div
                                className="modal-footer"
                        >
                                <button
                                        className="btn btn-info"
                                        data-dismiss="modal"
                                        id="cancelModalundefined"
                                        type="button"
                                >
                                        Cancel
                                </button>
                                <button
                                        className="btn btn-danger"
                                        onClick={[Function]}
                                        type="button"
                                >
                                        Delete
                                </button>
                        </div>
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal-dialog modal-confirm"
>
          <div
                    className="modal-content"
          >
                    <div
                              className="modal-header"
                    >
                              <h4
                                        className="modal-title"
                              >
                                        Are you sure?
                              </h4>
                              <Unknown />
                    </div>
                    <div
                              className="modal-body"
                    >
                              <p>
                                        Do you really want to delete 
                                        <strong />
                                         
                                        from the menu? This process cannot be undone.
                              </p>
                    </div>
                    <div
                              className="modal-footer"
                    >
                              <button
                                        className="btn btn-info"
                                        data-dismiss="modal"
                                        id="cancelModalundefined"
                                        type="button"
                              >
                                        Cancel
                              </button>
                              <button
                                        className="btn btn-danger"
                                        onClick={[Function]}
                                        type="button"
                              >
                                        Delete
                              </button>
                    </div>
          </div>
</div>,
        "className": "modal fade",
        "id": "deleteFromMenuModalundefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-content"
>
            <div
                        className="modal-header"
            >
                        <h4
                                    className="modal-title"
                        >
                                    Are you sure?
                        </h4>
                        <Unknown />
            </div>
            <div
                        className="modal-body"
            >
                        <p>
                                    Do you really want to delete 
                                    <strong />
                                     
                                    from the menu? This process cannot be undone.
                        </p>
            </div>
            <div
                        className="modal-footer"
            >
                        <button
                                    className="btn btn-info"
                                    data-dismiss="modal"
                                    id="cancelModalundefined"
                                    type="button"
                        >
                                    Cancel
                        </button>
                        <button
                                    className="btn btn-danger"
                                    onClick={[Function]}
                                    type="button"
                        >
                                    Delete
                        </button>
            </div>
</div>,
          "className": "modal-dialog modal-confirm",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="modal-header"
>
                <h4
                                className="modal-title"
                >
                                Are you sure?
                </h4>
                <Unknown />
</div>,
              <div
                className="modal-body"
>
                <p>
                                Do you really want to delete 
                                <strong />
                                 
                                from the menu? This process cannot be undone.
                </p>
</div>,
              <div
                className="modal-footer"
>
                <button
                                className="btn btn-info"
                                data-dismiss="modal"
                                id="cancelModalundefined"
                                type="button"
                >
                                Cancel
                </button>
                <button
                                className="btn btn-danger"
                                onClick={[Function]}
                                type="button"
                >
                                Delete
                </button>
</div>,
            ],
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="modal-title"
>
                    Are you sure?
</h4>,
                  <Unknown />,
                ],
                "className": "modal-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Are you sure?",
                    "className": "modal-title",
                  },
                  "ref": null,
                  "rendered": "Are you sure?",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Do you really want to delete 
                  <strong />
                   
                  from the menu? This process cannot be undone.
</p>,
                "className": "modal-body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Do you really want to delete ",
                    <strong />,
                    " ",
                    "from the menu? This process cannot be undone.",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Do you really want to delete ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "strong",
                  },
                  " ",
                  "from the menu? This process cannot be undone.",
                ],
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn btn-info"
                    data-dismiss="modal"
                    id="cancelModalundefined"
                    type="button"
>
                    Cancel
</button>,
                  <button
                    className="btn btn-danger"
                    onClick={[Function]}
                    type="button"
>
                    Delete
</button>,
                ],
                "className": "modal-footer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cancel",
                    "className": "btn btn-info",
                    "data-dismiss": "modal",
                    "id": "cancelModalundefined",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "className": "btn btn-danger",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal fade"
          id="deleteFromMenuModalundefined"
>
          <div
                    className="modal-dialog modal-confirm"
          >
                    <div
                              className="modal-content"
                    >
                              <div
                                        className="modal-header"
                              >
                                        <h4
                                                  className="modal-title"
                                        >
                                                  Are you sure?
                                        </h4>
                                        <Unknown />
                              </div>
                              <div
                                        className="modal-body"
                              >
                                        <p>
                                                  Do you really want to delete 
                                                  <strong />
                                                   
                                                  from the menu? This process cannot be undone.
                                        </p>
                              </div>
                              <div
                                        className="modal-footer"
                              >
                                        <button
                                                  className="btn btn-info"
                                                  data-dismiss="modal"
                                                  id="cancelModalundefined"
                                                  type="button"
                                        >
                                                  Cancel
                                        </button>
                                        <button
                                                  className="btn btn-danger"
                                                  onClick={[Function]}
                                                  type="button"
                                        >
                                                  Delete
                                        </button>
                              </div>
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-dialog modal-confirm"
>
            <div
                        className="modal-content"
            >
                        <div
                                    className="modal-header"
                        >
                                    <h4
                                                className="modal-title"
                                    >
                                                Are you sure?
                                    </h4>
                                    <Unknown />
                        </div>
                        <div
                                    className="modal-body"
                        >
                                    <p>
                                                Do you really want to delete 
                                                <strong />
                                                 
                                                from the menu? This process cannot be undone.
                                    </p>
                        </div>
                        <div
                                    className="modal-footer"
                        >
                                    <button
                                                className="btn btn-info"
                                                data-dismiss="modal"
                                                id="cancelModalundefined"
                                                type="button"
                                    >
                                                Cancel
                                    </button>
                                    <button
                                                className="btn btn-danger"
                                                onClick={[Function]}
                                                type="button"
                                    >
                                                Delete
                                    </button>
                        </div>
            </div>
</div>,
          "className": "modal fade",
          "id": "deleteFromMenuModalundefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="modal-content"
>
              <div
                            className="modal-header"
              >
                            <h4
                                          className="modal-title"
                            >
                                          Are you sure?
                            </h4>
                            <Unknown />
              </div>
              <div
                            className="modal-body"
              >
                            <p>
                                          Do you really want to delete 
                                          <strong />
                                           
                                          from the menu? This process cannot be undone.
                            </p>
              </div>
              <div
                            className="modal-footer"
              >
                            <button
                                          className="btn btn-info"
                                          data-dismiss="modal"
                                          id="cancelModalundefined"
                                          type="button"
                            >
                                          Cancel
                            </button>
                            <button
                                          className="btn btn-danger"
                                          onClick={[Function]}
                                          type="button"
                            >
                                          Delete
                            </button>
              </div>
</div>,
            "className": "modal-dialog modal-confirm",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="modal-header"
>
                  <h4
                                    className="modal-title"
                  >
                                    Are you sure?
                  </h4>
                  <Unknown />
</div>,
                <div
                  className="modal-body"
>
                  <p>
                                    Do you really want to delete 
                                    <strong />
                                     
                                    from the menu? This process cannot be undone.
                  </p>
</div>,
                <div
                  className="modal-footer"
>
                  <button
                                    className="btn btn-info"
                                    data-dismiss="modal"
                                    id="cancelModalundefined"
                                    type="button"
                  >
                                    Cancel
                  </button>
                  <button
                                    className="btn btn-danger"
                                    onClick={[Function]}
                                    type="button"
                  >
                                    Delete
                  </button>
</div>,
              ],
              "className": "modal-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="modal-title"
>
                      Are you sure?
</h4>,
                    <Unknown />,
                  ],
                  "className": "modal-header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Are you sure?",
                      "className": "modal-title",
                    },
                    "ref": null,
                    "rendered": "Are you sure?",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    Do you really want to delete 
                    <strong />
                     
                    from the menu? This process cannot be undone.
</p>,
                  "className": "modal-body",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Do you really want to delete ",
                      <strong />,
                      " ",
                      "from the menu? This process cannot be undone.",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Do you really want to delete ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "strong",
                    },
                    " ",
                    "from the menu? This process cannot be undone.",
                  ],
                  "type": "p",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="btn btn-info"
                      data-dismiss="modal"
                      id="cancelModalundefined"
                      type="button"
>
                      Cancel
</button>,
                    <button
                      className="btn btn-danger"
                      onClick={[Function]}
                      type="button"
>
                      Delete
</button>,
                  ],
                  "className": "modal-footer",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Cancel",
                      "className": "btn btn-info",
                      "data-dismiss": "modal",
                      "id": "cancelModalundefined",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Cancel",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Delete",
                      "className": "btn btn-danger",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Delete",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
